Client/src/Components/Charts/LineGraph/index.jsx

```diff
@@ -0,0 +1,132 @@
+import {
+	LineChart,
+	Line,
+	XAxis,
+	YAxis,
+	CartesianGrid,
+	Tooltip,
+	Legend,
+	ResponsiveContainer,
+} from "recharts";
+
+import PropTypes from "prop-types";
+import { useTheme } from "@emotion/react";
+import { Stack } from "@mui/material";
+import { useMemo } from "react";
+
+const LineGraph = ({
+	chartData = [],
+	chartConfig = {},
+	styleConfig = {},
+	onClickHandler,
+	onMouseEnterHandler,
+	onMouseMoveHandler,
+	onMouseLeaveHandler,
+}) => {
+	const theme = useTheme();
+
+	const memoizedChartData = useMemo(() => {
+		return chartData.length ? chartData : [];
+	}, [chartData]);
+
+	const { unit = "", metricName = "value" } = chartConfig;
+
+	const {
+		width = "100%",
+		height = 300,
+		chartStrokeColor = theme.palette.primary.main,
+		marginObject = { top: 5, right: 5, bottom: 5, left: 5 },
+	} = styleConfig;
+
+	if (!memoizedChartData.length) {
```

Comment:
Here we are using a check against 0. To make that explicit, I suggest we declare chartData.length ===0